crates/turborepo-lib/src/config/client.rs

```diff
@@ -42,18 +55,181 @@ impl ClientConfigLoader {
         self
     }
 
+    #[allow(dead_code)]
+    pub fn with_environment(mut self, environment: Option<HashMap<String, String>>) -> Self {
+        self.environment = environment;
+        self
+    }
+
     #[allow(dead_code)]
     pub fn load(self) -> Result<ClientConfig> {
         let Self {
             remote_cache_timeout,
+            environment,
         } = self;
 
-        let config: ClientConfigValue = Config::builder()
-            .add_source(Environment::with_prefix("turbo"))
+        let config_attempt: Result<ClientConfigValue, ConfigError> = Config::builder()
+            .set_default("remote_cache_timeout", DEFAULT_TIMEOUT)?
+            .add_source(Environment::with_prefix("turbo").source(environment))
             .set_override_option("remote_cache_timeout", remote_cache_timeout)?
             .build()?
-            .try_deserialize()?;
+            .try_deserialize();
+
+        // This goes wrong when TURBO_REMOTE_CACHE_TIMEOUT can't be deserialized to u64
+        match config_attempt {
+            Err(_) => Ok(ClientConfig {
```

Comment:
Should we log something here to inform users that we're not using the env variable?

Reply:
Yes, but presently we have no logging anywhere in our config modules. Given that, I'm not going to add this here because we should come up with a consistent strategy for this.

Reply:
TURBO-857